### PR TITLE
Fix crash with timer being doubly invalidated

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
 
     public var device: RileyLinkDevice!
     
-    var rssiFetchTimer: NSTimer!
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift:18]: Every time you write an exclamation mark, someone at Apple deletes a Radar. 😉

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 2]
```diff
@@ -82,14 +80,18 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
         if appeared {
             tableView.reloadData()
         }
+        
+        rssiFetchTimer = NSTimer.scheduledTimerWithTimeInterval(3, target: self, selector: #selector(updateRSSI), userInfo: nil, repeats: true)
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift:88]: `viewWillAppear` can run multiple times before `viewDidDisappear` runs, for example when cancelling a swift-from-left-edge gesture after starting it.

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift (hunk 1 above)
Comment (ps2) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift:18]: Yep.  Feel like I just got my hand smacked for reaching for the cookie jar. :)

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 3]
```diff
@@ -15,7 +15,7 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
 
     public var device: RileyLinkDevice!
     
-    var rssiFetchTimer: NSTimer!
+    var rssiFetchTimer: NSTimer?
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: Use `didSet` to invalidate, to make sure you're always calling `invalidate` before replacing the reference.

Comment (ps2) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: great idea.

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift (hunk 2 above)
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift:88]: But `viewWillDisappear` should be balanced with `viewWillAppear`, so let's set `rssiFetchTimer` to `nil` in that method.

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 4]
```diff
@@ -82,14 +80,18 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
...
+        rssiFetchTimer = NSTimer.scheduledTimerWithTimeInterval(3, target: self, selector: #selector(updateRSSI), userInfo: nil, repeats: true)
 
         appeared = true
     }
     
     public override func viewDidDisappear(animated: Bool) {
         super.viewDidDisappear(animated)
         
-        rssiFetchTimer.invalidate()
+        if let timer = rssiFetchTimer {
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: with the `didSet` change above, you can delete these three lines.

**Comment by loudnate** [comment]
I'm sure before long you'll hate `NSTimer` as much as I've told you I do 😄 

Here's a reference of how I use it, clunkily: https://github.com/loudnate/Loop/blob/master/Loop/Views/LoopCompletionHUDView.swift

**Comment by loudnate** [comment]
There is one more edge case where the app becomes inactive while the view controller is visible. I think it's fine to ignore for now.

**Comment by loudnate** [comment]
LGTM

**Comment by loudnate** [comment]
Can you send a PR to Loop with the updated frameworks after you release this? I'm away from a computer today.

**Comment by ps2** [comment]
sure

**Pull request merged**